iots-export/test/IOTS/fullContract.ts

```diff
@@ -57,40 +57,40 @@ const VestingTranche = t.type({
     validity: IntervalSlot
 });
 
-const FunctionAArgA = t.tuple([
-    CurrencySymbol,
-    TokenName
-]);
-
-const FunctionAArgB = t.union([
-    Value,
-    t.null
-]);
+const FunctionAArgs = t.type({
+    a: t.tuple([
+        CurrencySymbol,
+        TokenName
+    ]),
+    b: t.union([
+        Value,
+        t.null
+    ]),
+    c: IntervalSlot,
+    d: t.array(
+        VestingTranche
+    )
+});
 
-const FunctionAArgC = IntervalSlot;
+const FunctionAReturn = t.string;
 
-const FunctionAArgD = t.array(
-    VestingTranche
+export const FunctionA = createEndpoint<typeof FunctionAArgs, typeof FunctionAReturn, t.NullC>(
+    'FunctionA',
+    FunctionAArgs,
+    FunctionAReturn
 );
 
-const FunctionAArgReturn = t.string;
-
-type FunctionA = (
-    a: t.TypeOf<typeof FunctionAArgA>,
-    b: t.TypeOf<typeof FunctionAArgB>,
-    c: t.TypeOf<typeof FunctionAArgC>,
-    d: t.TypeOf<typeof FunctionAArgD>
-) => t.TypeOf<typeof FunctionAArgReturn>;
+const FunctionBArgs = t.type({
+    a: IntervalSlot,
+    b: t.array(
+        CurrencySymbol
+    )
+});
 
-const FunctionBArgA = IntervalSlot;
+const FunctionBReturn = t.string;
 
-const FunctionBArgB = t.array(
-    CurrencySymbol
+export const FunctionB = createEndpoint<typeof FunctionBArgs, typeof FunctionBReturn, t.NullC>(
```

Comment:
What is this `createEndpoint` thing?

Reply:
To quote Sam:

> To properly declare the contracts, I’ve create a createEndpoint method that generates nice types as well as an executable JS object. [...]  an import to this should not be in your string, as our compiler tooling will source this.

Reply:
It's just a bit weird that it's not declared here... you implicitly have to have this in scope for it to work.

Probably I just wish it was called something more generic so we could continue pretending that `iots-export` is a standalone thing. :sweat_smile: